framework/src/main/java/org/checkerframework/common/wholeprograminference/WholeProgramInferenceJavaParserStorage.java

```diff
@@ -706,64 +706,50 @@ public void processClass(ClassTree javacTree, AnnotationDeclaration javaParserNo
-              Element classElt = TreeUtils.elementFromDeclaration(body);
-              if (classElt != null) {
-                addClass(body, null);
-              } else {
-                // If such an element does not exist, compute the name of the class,
-                // instead.  This method of computing the name is not 100% guaranteed to
-                // be reliable, but it should be sufficient for WPI's purposes here: if
-                // the wrong name is computed, the worst outcome is a false positive
-                // because WPI inferred an untrue annotation.
-                @BinaryName String className;
-                if ("".contentEquals(body.getSimpleName())) {
-                  @SuppressWarnings("signature:assignment") // computed from string concatenation
-                  @BinaryName String computedName =
-                      javaParserClass.getFullyQualifiedName().get() + "$" + ++innerClassCount;
-                  className = computedName;
-                } else {
-                  @SuppressWarnings("signature:assignment") // computed from string concatenation
-                  @BinaryName String computedName =
-                      javaParserClass.getFullyQualifiedName().get()
-                          + "$"
-                          + body.getSimpleName().toString();
-                  className = computedName;
-                }
-                addClass(body, className, null);
-              }
+              addClass(body, null);
             }
           }
 
           /**
            * Creates a wrapper around the class for {@code tree} and stores it in {@code
            * sourceAnnos}.
            *
-           * <p>This method assumes that there is an Element corresponding to {@code tree} if {@code
-           * classNameKey} is null.
+           * <p>This method computes the name of the class and uses it as the key for {@code
+           * classToAnnos}
            *
            * @param tree tree to add. Its corresponding element is used as the key for {@code
            *     classToAnnos} if {@code classNameKey} is null.
-           * @param classNameKey if non-null, used as the key for {@code classToAnnos} instead of
-           *     the binary name of the element corresponding to {@code tree}
            * @param javaParserNode the node corresponding to the declaration, which is used to place
            *     annotations on the class itself. Can be null, e.g. for an anonymous class.
            */
-          private void addClass(
-              ClassTree tree,
-              @Nullable @BinaryName String classNameKey,
-              @Nullable TypeDeclaration<?> javaParserNode) {
+          private void addClass(ClassTree tree, @Nullable TypeDeclaration<?> javaParserNode) {
             String className;
-            if (classNameKey != null) {
-              className = classNameKey;
+            // elementFromTree returns null instead of crashing when no element exists
```

Comment:
This calls `elementFromDeclaration`, not `elementFromTree`. (This might be because `elementFromDeclaration` used to be called `elementFromTree`, but you should double-check that the right method is being called and then fix this comment if it is out-of-date.)